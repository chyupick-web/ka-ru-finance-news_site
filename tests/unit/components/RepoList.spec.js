import { shallowMount, createLocalVue } from '@vue/test-utils'
import Component from '@/components/RepoList.vue'
import BootstrapVue from 'bootstrap-vue'

const localVue = createLocalVue()
localVue.use(BootstrapVue)

describe('RepoList.vue', () => {
  let propsData

  beforeEach(() => {
    propsData = {
      items: [
        {
          owner: '<OWNER>',
          ownerUrl: 'https://github.com/<OWNER>',
          name: '<NAME>',
          url: 'https://github.com/<OWNER>/<NAME>',
          description: 'Nulla quis lorem ut libero malesuada feugiat.',
          stars: 150,
          forks: 25,
          language: {
            title: null,
            color: null
          }
        },
        {
          owner: '<OWNER 2>',
          ownerUrl: 'https://github.com/<OWNER 2>',
          name: '<NAME 2>',
          url: 'https://github.com/<OWNER 2>/<NAME 2>',
          description: 'Cras ultricies ligula sed magna dictum porta.',
          stars: 40,
          forks: 12,
          language: {
            title: 'Javascript',
            color: '#b00'
          }
        }
      ]
    }
  })

  it('matches its default props if none are set', () => {
    const wrapper = shallowMount(Component, { localVue })
    expect(wrapper.props()).toEqual({
      title: 'All Repos',
      colWidth: '8',
      items: [],
      filter: false,
      showTitleCount: false,
      sort: false
    })
  })

  it('matches snapshot with default props', () => {
    const wrapper = shallowMount(Component, { localVue })
    expect(wrapper.html()).toMatchSnapshot('defaultProps')
  })

  it('matches snapshot with set props (no filter, no title count, default title)', () => {
    const wrapper = shallowMount(Component, { localVue, propsData })
    expect(wrapper.html()).toMatchSnapshot('noFilterNoTitleCountDefaultTitle')
  })

  it('matches snapshot with set props (with filter, with title count, 4 columns, set title)', () => {
    propsData.filter = true
    propsData.showTitleCount = true
    propsData.colWidth = '4'
    propsData.title = 'Featured Repos'
    const wrapper = shallowMount(Component, { localVue, propsData })
    expect(wrapper.html()).toMatchSnapshot('withFilterWithTitleCountFourColumnsWithSetTitle')
  })

  it('filters by language', () => {
    propsData.filter = true
    const wrapper = shallowMount(Component, { localVue, propsData })
    wrapper.setData({ text: 'javascri' })
    expect(wrapper.html()).toMatchSnapshot('FilterByLanguage')
  })
})
