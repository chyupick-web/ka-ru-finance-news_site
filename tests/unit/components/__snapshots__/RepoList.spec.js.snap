// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList.vue filters by language: FilterByLanguage 1`] = `
<b-container-stub tag="div" fluid="true" class="pt-4 pb-2">
  <b-row-stub tag="div">
    <b-col-stub tag="div">
      <b-container-stub tag="div">
        <b-row-stub tag="div">
          <b-col-stub tag="div">
            <h3 class="mb-4">
              All Repos

            </h3>
            <b-row-stub tag="div" class="mb-3">
              <b-col-stub md="6" tag="div">
                <b-form-input-stub value="javascri" autocomplete="off" placeholder="Filter by title or language" type="text"></b-form-input-stub>
              </b-col-stub>
            </b-row-stub>
            <b-row-stub tag="div">
              <b-col-stub lg="8" tag="div" class="mb-4">
                <repo-card-stub item="[object Object]"></repo-card-stub>
              </b-col-stub>
            </b-row-stub>
          </b-col-stub>
        </b-row-stub>
      </b-container-stub>
    </b-col-stub>
  </b-row-stub>
</b-container-stub>
`;

exports[`RepoList.vue matches snapshot with default props: defaultProps 1`] = `
<b-container-stub tag="div" fluid="true" class="pt-4 pb-2">
  <b-row-stub tag="div">
    <b-col-stub tag="div">
      <b-container-stub tag="div">
        <b-row-stub tag="div">
          <b-col-stub tag="div">
            <h3 class="mb-4">
              All Repos

            </h3>
            <!---->
            <b-row-stub tag="div"></b-row-stub>
          </b-col-stub>
        </b-row-stub>
      </b-container-stub>
    </b-col-stub>
  </b-row-stub>
</b-container-stub>
`;

exports[`RepoList.vue matches snapshot with set props (no filter, no title count, default title): noFilterNoTitleCountDefaultTitle 1`] = `
<b-container-stub tag="div" fluid="true" class="pt-4 pb-2">
  <b-row-stub tag="div">
    <b-col-stub tag="div">
      <b-container-stub tag="div">
        <b-row-stub tag="div">
          <b-col-stub tag="div">
            <h3 class="mb-4">
              All Repos

            </h3>
            <!---->
            <b-row-stub tag="div">
              <b-col-stub lg="8" tag="div" class="mb-4">
                <repo-card-stub item="[object Object]"></repo-card-stub>
              </b-col-stub>
              <b-col-stub lg="8" tag="div" class="mb-4">
                <repo-card-stub item="[object Object]"></repo-card-stub>
              </b-col-stub>
            </b-row-stub>
          </b-col-stub>
        </b-row-stub>
      </b-container-stub>
    </b-col-stub>
  </b-row-stub>
</b-container-stub>
`;

exports[`RepoList.vue matches snapshot with set props (with filter, with title count, 4 columns, set title): withFilterWithTitleCountFourColumnsWithSetTitle 1`] = `
<b-container-stub tag="div" fluid="true" class="pt-4 pb-2">
  <b-row-stub tag="div">
    <b-col-stub tag="div">
      <b-container-stub tag="div">
        <b-row-stub tag="div">
          <b-col-stub tag="div">
            <h3 class="mb-4">
              Featured Repos
              (2)
            </h3>
            <b-row-stub tag="div" class="mb-3">
              <b-col-stub md="6" tag="div">
                <b-form-input-stub value="" autocomplete="off" placeholder="Filter by title or language" type="text"></b-form-input-stub>
              </b-col-stub>
            </b-row-stub>
            <b-row-stub tag="div">
              <b-col-stub lg="4" tag="div" class="mb-4">
                <repo-card-stub item="[object Object]"></repo-card-stub>
              </b-col-stub>
              <b-col-stub lg="4" tag="div" class="mb-4">
                <repo-card-stub item="[object Object]"></repo-card-stub>
              </b-col-stub>
            </b-row-stub>
          </b-col-stub>
        </b-row-stub>
      </b-container-stub>
    </b-col-stub>
  </b-row-stub>
</b-container-stub>
`;
